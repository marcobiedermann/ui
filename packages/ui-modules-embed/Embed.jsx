import classNames from 'classnames';
import PropTypes from 'prop-types';
import React from 'react';
import styles from './embed.css';

const Embed = ({ children, aspectRatio }) => (
  <div
    className={classNames(styles.embed, {
      [styles[`embed--${aspectRatio}`]]: aspectRatio,
    })}
  >
    {children}
  </div>
);

Embed.propTypes = {
  aspectRatio: PropTypes.string,
  children: PropTypes.node,
};

Embed.defaultProps = {
  aspectRatio: '',
  children: null,
};

export default Embed;
