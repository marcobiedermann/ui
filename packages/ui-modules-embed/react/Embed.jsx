import classNames from 'classnames';
import PropTypes from 'prop-types';
import React, { PureComponent } from 'react';
import styles from '../embed.css';

class Embed extends PureComponent {
  static propTypes = {
    aspectRatio: PropTypes.string,
    children: PropTypes.node,
  };

  static defaultProps = {
    aspectRatio: '',
    children: null,
  };

  render() {
    const { children, aspectRatio } = this.props;

    return (
      <div
        className={classNames(styles.embed, {
          [styles[`embed--${aspectRatio}`]]: aspectRatio,
        })}
      >
        {children}
      </div>
    );
  }
}

export default Embed;
