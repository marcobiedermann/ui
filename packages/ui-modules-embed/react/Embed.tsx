import classNames from 'classnames';
import React, { FC } from 'react';
import styles from '../embed.css';

export interface EmbedProps {
  aspectRatio?: '4-3' | '16-9';
}

export const Embed: FC = (props) => {
  const { aspectRatio = '16-9', ...otherProps } = props;

  return (
    <div
      className={classNames(styles.embed, {
        [styles[`embed--${aspectRatio}`]]: aspectRatio,
      })}
      {...otherProps}
    />
  );
};

export default Embed;
