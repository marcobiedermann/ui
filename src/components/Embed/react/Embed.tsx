import classNames from 'classnames';
import React, { FC, HTMLAttributes } from 'react';
import styles from '../embed.css';

export interface EmbedProps extends HTMLAttributes<HTMLDivElement> {
  aspectRatio?: '4-3' | '16-9';
}

const Embed: FC<EmbedProps> = (props: EmbedProps) => {
  const { aspectRatio = '16-9', className, ...otherProps } = props;

  return (
    <div
      className={classNames(className, styles.embed, {
        [styles[`embed--${aspectRatio}`]]: aspectRatio,
      })}
      {...otherProps}
    />
  );
};

export default Embed;
