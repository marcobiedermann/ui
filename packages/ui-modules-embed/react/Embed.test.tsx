import React from 'react';
import renderer from 'react-test-renderer';
import Embed from './Embed';

describe('Embed component', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <Embed>
          <iframe src="/" title="@marcobiedermann/ui" />
        </Embed>,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders aspect ratio correctly', () => {
    const tree = renderer
      .create(
        <Embed aspectRatio="16-9">
          <iframe src="/" title="@marcobiedermann/ui" />
        </Embed>,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
