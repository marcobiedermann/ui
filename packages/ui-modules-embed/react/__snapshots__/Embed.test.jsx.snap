// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Embed component renders aspect ratio correctly 1`] = `
<div
  className="embed embed--16-9"
>
  <iframe
    src="/"
    title="@marcobiedermann/ui"
  />
</div>
`;

exports[`Embed component renders correctly 1`] = `
<div
  className="embed"
>
  <iframe
    src="/"
    title="@marcobiedermann/ui"
  />
</div>
`;
